In [12]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [13]:
USDA = pd.read_csv('https://storage.googleapis.com/dimensionless/Analytics/USDA.csv')

In [14]:
print(type(USDA))
USDA.head()

<class 'pandas.core.frame.DataFrame'>


ID               Description  Calories  Protein  TotalFat  Carbohydrate  \
0  1001          BUTTER,WITH SALT     717.0     0.85     81.11          0.06   
1  1002  BUTTER,WHIPPED,WITH SALT     717.0     0.85     81.11          0.06   
2  1003      BUTTER OIL,ANHYDROUS     876.0     0.28     99.48          0.00   
3  1004               CHEESE,BLUE     353.0    21.40     28.74          2.34   
4  1005              CHEESE,BRICK     371.0    23.24     29.68          2.79   

   Sodium  SaturatedFat  Cholesterol  Sugar  Calcium  Iron  Potassium  \
0   714.0        51.368        215.0   0.06     24.0  0.02       24.0   
1   827.0        50.489        219.0   0.06     24.0  0.16       26.0   
2     2.0        61.924        256.0   0.00      4.0  0.00        5.0   
3  1395.0        18.669         75.0   0.50    528.0  0.31      256.0   
4   560.0        18.764         94.0   0.51    674.0  0.43      136.0   

   VitaminC  VitaminE  VitaminD  
0       0.0      2.32       1.5  
1       0.0      2.32       1.5  
2       0.0      2.80       1.8  
3       0.0      0.25       0.5  
4       0.0      0.26       0.5

<h3>1. Obtain high-level statistical information about your dataset.
( Mean, Median, SD, Q1, Q3 etc)</h3>

In [57]:
USDA.describe()

ID     Calories      Protein     TotalFat  Carbohydrate  \
count   7058.000000  7057.000000  7057.000000  7057.000000   7057.000000   
mean   14259.821196   219.695338    11.710368    10.320614     20.697860   
std     8577.179705   172.198755    10.919356    16.814191     27.630443   
min     1001.000000     0.000000     0.000000     0.000000      0.000000   
25%     8387.250000    85.000000     2.290000     0.720000      0.000000   
50%    13293.500000   181.000000     8.200000     4.370000      7.130000   
75%    18336.750000   331.000000    20.430000    12.700000     28.170000   
max    93600.000000   902.000000    88.320000   100.000000    100.000000   

             Sodium  SaturatedFat  Cholesterol        Sugar      Calcium  \
count   6974.000000   6757.000000  6770.000000  5148.000000  6922.000000   
mean     322.059220      3.452267    41.551994     8.256540    73.530627   
std     1045.416931      6.921267   122.963028    15.361509   222.445338   
min        0.000000      0.000000     0.000000     0.000000     0.000000   
25%       37.000000      0.172000     0.000000     0.000000     9.000000   
50%       79.000000      1.256000     3.000000     1.395000    19.000000   
75%      386.000000      4.028000    69.000000     7.875000    56.000000   
max    38758.000000     95.600000  3100.000000    99.800000  7364.000000   

              Iron     Potassium     VitaminC     VitaminE     VitaminD  
count  6935.000000   6649.000000  6726.000000  4338.000000  4224.000000  
mean      2.828368    301.357949     9.435980     1.487462     0.576918  
std       6.019878    415.638949    71.256536     5.386914     4.301147  
min       0.000000      0.000000     0.000000     0.000000     0.000000  
25%       0.520000    135.000000     0.000000     0.120000     0.000000  
50%       1.330000    250.000000     0.000000     0.270000     0.000000  
75%       2.620000    348.000000     3.100000     0.710000     0.100000  
max     123.600000  16500.000000  2400.000000   149.400000   250.000000

<h3>2. The maximum level of Sodium is 38,758 milligrams, which is very high. Find out which food this
corresponds to?</h3>

In [9]:
print(USDA.Sodium.max())
USDA.Description[USDA.Sodium.idxmax()]

38758.0


'SALT,TABLE'

<h3>3. Find out which foods contain more than 10,000 milligrams of sodium and store it to a new data
frame, “HighSodium”.</h3>

In [18]:
HighSodium=USDA[USDA.Sodium>10000]
HighSodium

ID                                        Description  Calories  \
264    2047                                         SALT,TABLE       0.0   
921    6075                  SOUP,BF BROTH OR BOUILLON,PDR,DRY     213.0   
922    6076                          SOUP,BEEF BROTH,CUBED,DRY     170.0   
924    6080                   SOUP,CHICK BROTH OR BOUILLON,DRY     267.0   
925    6081                         SOUP,CHICK BROTH CUBES,DRY     198.0   
937    6115                                   GRAVY,AU JUS,DRY     313.0   
1302   6979                                       ADOBO FRESCO     271.0   
5320  18369  LEAVENING AGENTS,BAKING PDR,DOUBLE-ACTING,NA A...      53.0   
5323  18372                       LEAVENING AGENTS,BAKING SODA       0.0   
5697  19225                    DESSERTS,RENNIN,TABLETS,UNSWTND      84.0   

      Protein  TotalFat  Carbohydrate   Sodium  SaturatedFat  Cholesterol  \
264      0.00      0.00          0.00  38758.0         0.000          0.0   
921     15.97      8.89         17.40  26000.0         4.320         10.0   
922     17.30      4.00         16.10  24000.0         1.990          4.0   
924     16.66     13.88         18.01  23875.0         3.430         13.0   
925     14.60      4.70         23.50  24000.0         1.200         13.0   
937      9.20      9.63         47.49  11588.0         2.026          4.0   
1302     2.00     20.90         18.60  17152.0         2.910          0.0   
5320     0.00      0.00         27.70  10600.0         0.000          0.0   
5323     0.00      0.00          0.00  27360.0         0.000          0.0   
5697     1.00      0.10         19.80  26050.0         0.041          0.0   

      Sugar  Calcium   Iron  Potassium  VitaminC  VitaminE  VitaminD  
264    0.00     24.0   0.33        8.0       0.0      0.00       0.0  
921   16.71     60.0   1.00      446.0       0.0      2.17       0.0  
922   14.51     60.0   2.23      403.0       0.0      0.00       0.0  
924   17.36    187.0   1.03      309.0       1.1      0.46       0.0  
925    0.00    190.0   1.87      374.0       1.0      0.09       NaN  
937     NaN    140.0   9.30      279.0       1.0       NaN       NaN  
1302   2.03    123.0   3.20      187.0       5.8      2.69       0.0  
5320   0.00   5876.0  11.02       20.0       0.0      0.00       0.0  
5323   0.00      0.0   0.00        0.0       0.0      0.00       0.0  
5697    NaN   3733.0   7.07      292.0       0.0       NaN       NaN

<h3>4. Do we have “caviar” present in "HighSodium" dataframe?</h3>

In [19]:
HighSodium[HighSodium['Description']=='CAVIAR'].Sodium

Series([], Name: Sodium, dtype: float64)

<h3>5. How much sodium content is present in 100 grams of "caviar"?</h3>

In [30]:
for x in USDA.Description:
    if x.lower()=='caviar':
        print(x)
        print(USDA[USDA['Description']==x].Sodium)
        
#USDA[USDA['Description']=='CAVIAR'].Sodium

CAVIAR
4153    1500.0
Name: Sodium, dtype: float64


<h3>6. Add a variable to our USDA dataframe that takes a value 1 if the food has higher sodium than
average,and 0 if the food has lower sodium than average.</h3>

In [75]:
USDA['Value']=[1 if x>USDA.Sodium.mean() else 0 for x in USDA.Sodium]
USDA.head()

ID               Description  Calories  Protein  TotalFat  Carbohydrate  \
0  1001          BUTTER,WITH SALT     717.0     0.85     81.11          0.06   
1  1002  BUTTER,WHIPPED,WITH SALT     717.0     0.85     81.11          0.06   
2  1003      BUTTER OIL,ANHYDROUS     876.0     0.28     99.48          0.00   
3  1004               CHEESE,BLUE     353.0    21.40     28.74          2.34   
4  1005              CHEESE,BRICK     371.0    23.24     29.68          2.79   

   Sodium  SaturatedFat  Cholesterol  Sugar  Calcium  Iron  Potassium  \
0   714.0        51.368        215.0   0.06     24.0  0.02       24.0   
1   827.0        50.489        219.0   0.06     24.0  0.16       26.0   
2     2.0        61.924        256.0   0.00      4.0  0.00        5.0   
3  1395.0        18.669         75.0   0.50    528.0  0.31      256.0   
4   560.0        18.764         94.0   0.51    674.0  0.43      136.0   

   VitaminC  VitaminE  VitaminD  Value  
0       0.0      2.32       1.5      1  
1       0.0      2.32       1.5      1  
2       0.0      2.80       1.8      0  
3       0.0      0.25       0.5      1  
4       0.0      0.26       0.5      1

<h3>7. How many foods have higher sodium level than average?</h3>

In [78]:
sum(USDA.Sodium>USDA.Sodium.mean())

2090

<h3>8. How many foods have both high sodium and high fat level than average?</h3>

In [15]:
count=0
j=0
for i in USDA.Sodium:
    if i>USDA.Sodium.mean():
        if (USDA.TotalFat[j]>USDA.TotalFat.mean()):
                count=count+1
                j=j+1            
print(count)

11


In [10]:
range(len(USDA.TotalFat))

range(0, 7058)

<h3>9. Compute the average amount of iron sorted by high and low protein?</h3>

<h3>10. Find the maximum level of vitamin C in foods with high and low carbs?</h3>